system/include/webgpu/webgpu.h

```diff
@@ -846,12 +945,11 @@ extern "C" {
 #endif
 
 typedef void (*WGPUBufferMapCallback)(WGPUBufferMapAsyncStatus status, void * userdata);
+typedef void (*WGPUCreateReadyComputePipelineCallback)(WGPUCreateReadyPipelineStatus status, WGPUComputePipeline pipeline, char const * message, void * userdata);
+typedef void (*WGPUCreateReadyRenderPipelineCallback)(WGPUCreateReadyPipelineStatus status, WGPURenderPipeline pipeline, char const * message, void * userdata);
 typedef void (*WGPUDeviceLostCallback)(char const * message, void * userdata);
 typedef void (*WGPUErrorCallback)(WGPUErrorType type, char const * message, void * userdata);
 typedef void (*WGPUFenceOnCompletionCallback)(WGPUFenceCompletionStatus status, void * userdata);
-typedef void (*WGPURequestAdapterCallback)(WGPUAdapter result, void * userdata);
-typedef void (*WGPURequestDeviceCallback)(WGPUDevice result, void * userdata);
-typedef void (*WGPUSurfaceGetPreferredFormatCallback)(WGPUTextureFormat format, void * userdata);
```

Comment:
Neither Dawn nor Emscripten has implemented these, but we should. Emscripten in particular should implement them so we can kill off the preinitializedWebGPUDevice thing. Can you leave them in even though they still won't be implemented?

Reply:
Yep, I somehow missed this comment because github hides a bunch of comments. Those are back in